server/plugins/oidc/src/main/java/com/walmartlabs/concord/server/plugins/oidc/OidcRealm.java

```diff
@@ -84,6 +91,26 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
 
         userManager.update(u.getId(), profile.getDisplayName(), profile.getEmail(), null, false, null);
 
+        Set<UUID> newTeams = new HashSet<>();
+        teamDao.tx(tx -> {
+            List<UserTeam> currentTeams = userDao.listTeams(tx, u.getId());
+
+            for (Map.Entry<UUID, PluginConfiguration.TeamMapping> e : cfg.getTeamMapping().entrySet()) {
+                if (match(profile, e.getValue().getSources())) {
+                    if (!hasTeam(e.getKey(), e.getValue().getRole(), currentTeams)) {
+                        teamDao.upsertUser(tx, e.getKey(), u.getId(), e.getValue().getRole());
+                    }
+                    newTeams.add(e.getKey());
+                }
+            }
+
+            List<UUID> toRemove = currentTeams.stream()
+                    .map(UserTeam::teamId)
+                    .filter(o -> !newTeams.contains(o))
+                    .collect(Collectors.toList());
+            userDao.excludeFromTeams(tx, u.getId(), toRemove);
```

Comment:
Skip calling `excludeFromTeams`  if `toRemove` is empty maybe?

Reply:
oops. fixed.